### ISSUE TITLE
[Grid - Scrolling] Scrollable elements must have keyboard focusable content

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2895]
### Provide a general summary of the issue here

Some scrollable variants do not have keyboard focus or a keyboard focusable element inside them. While many browsers create a focus when CSS overflow is used, some do not and there are inconsistencies in even the browsers that do. Setting keyboard focus using tabindex="0" to the overflowing element (or having a focusable element inside the overflowing area) ensures that the element will be focusable even when browser does not automatically apply focus.

### 🤔 Expected Behavior?

Users are able to focus the table or its contents to use the scrolling function.

### 😯 Current Behavior

Users might not be able to use the scrolling functions because the table body might not be focusable if there are no focusable elements inside. 

### 💁 Possible Solution

Add tabindex="0" to <tbody> if it does not contain any focusable elements. 

### 🖥️ Steps to Reproduce

Use a variant with scrolling (e.g., Basics Ellipsize text, Scrolling any variant)
Use a browser which does not have this feature enabled (e.g. Safari, most chromium based ver. 132 and below).
1. Use keyboard to move to content which has a scrolling bar
2. If content has focusable elements (e.g., link, checkbox) you can use arrow keys to scroll the content 

### What browsers are you seeing the problem on? (only for bugs in code)

Safari, Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by a11ymiko** [comment]
As Valtteri suggested we should add `tabindex=0` to each grid that: 
1. is scrollable AND
2. has no interactive content inside 
because automatically added by browser `tabindex` to tables with scroll is not universally supported in all browsers the same way.

`tabindex='0'` should be added to `<tbody>` only for grids that have scrollable content and inside of it there is no interactive elements like buttons or links. 
Additionally to `<tbody 'tabindex="0"'>` we should also add `aria-label='scrollable table'` to make it announced by some screen readers. Currently this announcing is well supported by VoiceOver but not by NVDA. However having this `aria-label` added to each scrollable table may make it more accessible for NVDA users once this tool will support that announcement the same way VoiceOver does it right now.
